### ISSUE TITLE
Wrapped FunctionOp

### ISSUE THREAD

**Issue by dietzc** [imagej/imagej-ops#231]
Hi @ctrueden,

there is some problem with `FunctionOp`s which are discovered using the `OpService`. Assume that I try to resolve a `FunctionOp` based on an input of type `A` and some output of type `O`. The `OpService` finds an `FunctionOp` `B to O` as there exists a converter from `A` to `B`. However, the resulting `FunctionOp` will only accept `A`s in the `compute(A in)`. Shouldn't the `OpService` wrap the orginal `FunctionOp`, such that the wrapped `FunctionOp` uses a converter inside the `compute(B...)`.

**Comment by ctrueden** [comment]
Hmm, good idea.

There may be performance issues with the wrapper calling into the `ConvertService` every time `compute` is called, though. So we may want to also cache the `Converter` instance used.

**Comment by dietzc** [comment]
which could be done in the initialize of the wrapped op?

**Comment by ctrueden** [comment]
The wrapper will need to do the caching, probably.

**Comment by dietzc** [comment]
https://github.com/imagej/imagej-ops/commit/1afa9f001548f9fbac087cdc99a23821bffbdd7f Wouldn't this be enough, if the `OpService` instantiates this `Op`?

**Comment by ctrueden** [comment]
Yeah. But I wouldn't label the worker `FunctionOp` or `Converter` with `@Parameter`, since the `ConvertedFunctionOp` is not actually declared as an `Op` plugin.

And the framework still needs to instantiate the `ConvertedFunctionOp`, which requires determining and setting the associated `Converter`. Probably, the [`specialOp` method](https://github.com/imagej/imagej-ops/blob/a83d59247c16637eb4228c6edf29ce2d12fe4f20/src/main/java/net/imagej/ops/AbstractOpEnvironment.java#L742-L750)) should be revised to support this.

**Comment by dietzc** [comment]
agreed. You always would have to create a `ConvertedFunctionOp`or `ConvertedComputerOp` if the special input of the `FunctionOp` or `ComputerOp` is converted. This would make the complete system more consistent.

**Comment by dietzc** [comment]
added test-case demonstrating the problem

https://github.com/imagej/imagej-ops/tree/converter-bug
